integration-tests/pgevent/src/main/java/org/apache/camel/quarkus/component/pgevent/it/PgeventRoutes.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.pgevent.it;
+
+import javax.inject.Named;
+import javax.ws.rs.Produces;
+
+import com.impossibl.postgres.jdbc.PGDataSource;
+import org.apache.camel.builder.RouteBuilder;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+public class PgeventRoutes extends RouteBuilder {
+    @ConfigProperty(name = "database.host")
+    String host;
+    @ConfigProperty(name = "database.port")
+    Integer port;
+    @ConfigProperty(name = "database.name")
+    String databaseName;
+    @ConfigProperty(name = "database.user")
+    String user;
+    @ConfigProperty(name = "database.password")
+    String password;
+
+    @Override
+    public void configure() throws Exception {
+        // producer for simple pub-sub
+        from("direct:pgevent-pub")
+                .to("pgevent://{{database.host}}:{{database.port}}/{{database.name}}/testchannel?user={{database.user}}&pass={{database.password}}");
+
+        //consumer for simple pub-sub
+        from("pgevent://{{database.host}}:{{database.port}}/{{database.name}}/testchannel?user={{database.user}}&pass={{database.password}}")
+                .log("Message got ${body}")
+                .bean(MyBean.class);
+
+        // producer with datasource
+        from("direct:pgevent-datasource")
+                .to("pgevent:///postgres/testchannel?datasource=#pgDataSource");
+
+        // consumer with datasource
+        from("pgevent:///postgres/testchannel?datasource=#pgDataSource")
+                .log("Message got ${body}")
+                .bean(MyBean.class);
+    }
+
+    @Produces
+    @Named("pgDataSource")
+    public PGDataSource loadDataSource() throws Exception {
+        PGDataSource dataSource = new PGDataSource();
```

Comment:
Maybe it'd be good to leverage the [Quarkus agroal](https://quarkus.io/guides/datasource) extension to create a named postgresql datasource? Or at least add another route / test case for that scenario. It's a good way of proving our stuff integrates well with Quarkus.

Reply:
good point!!

Reply:
@jamesnetherton well it's not the same datasource. This is datasource expected is the one from pgevent API. The one having a connection that enables us to have those events.

Reply:
if we want to have the agroal datasource, we need to convert this one to com.impossibl.postgres.jdbc.PGDatasource

Reply:
Ok, maybe we merge the PR as-is and create a follow up issue to look into this afterwards.

I think it should be possible to do. Both `PGDataSource` and `AgroalDatasource` extend `javax.sql.DataSource`. So doing `datasource=#nameOfTheAgroalDatasource` should work (in theory).

Reply:
@jamesnetherton yes, I've tested and it works fine. But in our camel component we get the connection from that datasource and if it's not an instance of PGConnection, we raise an error. 
We need to handle a connection of type java.sql.Connection first here : https://github.com/apache/camel/blob/d23efa9cf87f6f5dab886206a3c00cc5e0e2472c/components/camel-pgevent/src/main/java/org/apache/camel/component/pgevent/PgEventHelper.java#L35
So maybe follow up an issue for an improvement.

Reply:
I've created the issue : https://github.com/apache/camel-quarkus/issues/1909, to follow up